knowledge-base/grid-state-filter-nullable-bool.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Programatically filter a nullable bool column in the Grid by null values
+description: Learn how to programmatically filter nullable bool values in the Grid component for Blazor.
+type: how-to
+page_title: How to filter a Grid column bound to a nullable bool by the null values programatically
+slug: grid-kb-state-filter-nullable-bool
+tags: grid, blazor, column, datagrid, filter, state, nullable, bool
+res_type: kb
+ticketid: 1658561
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+I want to programatically filter a Grid column that is bound to `bool?` by the `null` values. 
+
+
+This KB article also answers the following questions:
+- How to filter a Grid column bound to a `bool?` by the null values programatically
+
+## Solution
+
+To filter a Grid column bound to a `bool?` by the null values programatically:
+
+````CSHTML
+@using Telerik.DataSource;
+
+<TelerikButton ThemeColor="primary" OnClick="@SetGridFilter">set filtering from code</TelerikButton>
+
+<TelerikGrid Data="@MyData" Height="400px" @ref="@GridRef"
+             Pageable="true" FilterMode="@GridFilterMode.FilterMenu">
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Id))" Width="120px" />
+        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" />
+        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
+        <GridColumn Field="@(nameof(SampleData.IsOnLeave))" Title="On Leave" />
+        <GridColumn Field="@(nameof(SampleData.HireDate))" Title="Hire Date" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private TelerikGrid<SampleData> GridRef { get; set; }
+
+
+    private async Task SetGridFilter()
+    {
+        GridState<SampleData> desiredState = new GridState<SampleData>()
+            {
+                FilterDescriptors = new List<IFilterDescriptor>()
+            {
+                new CompositeFilterDescriptor()
+                {
+                    FilterDescriptors = new FilterDescriptorCollection()
+                    {
+                        // Use the IsNull filter operator when filtering by null values.
+                        new FilterDescriptor() { Member = "IsOnLeave", Operator = FilterOperator.IsNull, Value = null, MemberType = typeof(bool?) },
+                    }
+                }
+            }
+            };
+
+        await GridRef.SetStateAsync(desiredState);
+    }
+
+    private IEnumerable<SampleData> MyData { get; set; }
+
+
+    protected override void OnInitialized()
+    {
+        Random random = new Random();
```

Comment:
Starting from .NET 6, it's recommended to use the static `Random.Shared` instead of `new Random()`.
https://andrewlock.net/building-a-thread-safe-random-implementation-for-dotnet-framework/